Lib/statistics.py

```diff
@@ -795,22 +787,34 @@ def _ss(data, c=None):
     """
     if c is not None:
         T, total, count = _sum((d := x - c) * d for x in data)
-        return (T, total)
-    T, total, count = _sum(data)
-    mean_n, mean_d = (total / count).as_integer_ratio()
-    partials = Counter()
-    for n, d in map(_exact_ratio, data):
-        diff_n = n * mean_d - d * mean_n
-        diff_d = d * mean_d
-        partials[diff_d * diff_d] += diff_n * diff_n
-    if None in partials:
+        return (T, total, count)
+    count = 0
+    sx_partials = {}
+    sx_partials_get = sx_partials.get
+    sxx_partials = {}
```

Comment:
Consider making `sxx_partials` and `sx_partials` instances of `defaultdict(int)` instead? Then the `..._get` bound methods wouldn't be needed (phrases like `= sx_partials_get(d, 0) +` could be deleted, replaced by `+=`).

Reply:
I looked at defaultdict and Counter.  Both looked nicer but were slower and they were inconsistent with Steven's other code in _sum().  So I left the code pretty much as he already laid it out and instead focused on the algorithmic improvement.

Defaultdict does pretty well when it has hits.  But with misses, `int()` is probably the worse way to create an integer.